packages/hash/api/src/model/knowledge/block.model.ts

```diff
@@ -0,0 +1,182 @@
+import { GraphApi } from "@hashintel/hash-graph-client";
+import { EntityModel, BlockModel, EntityModelCreateParams } from "..";
+import { WORKSPACE_TYPES } from "../../graph/workspace-types";
+import { workspaceAccountId } from "../util";
+
+type BlockModelCreateParams = Omit<
+  EntityModelCreateParams,
+  "properties" | "entityTypeModel"
+> & {
+  componentId: string;
+  blockData: EntityModel;
+};
+
+/**
+ * @class {@link BlockModel}
+ */
+export default class extends EntityModel {
+  static fromEntityModel(entity: EntityModel): BlockModel {
+    if (
+      entity.entityTypeModel.schema.$id !==
+      WORKSPACE_TYPES.entityType.block.schema.$id
+    ) {
+      throw new Error(
+        `Entity with id ${entity.entityId} is not a workspace block`,
+      );
+    }
+
+    return new BlockModel(entity);
+  }
+
+  /**
+   * Get a workspace block entity by its entity id.
+   *
+   * @param params.entityId - the entity id of the block
+   */
+  static async getBlockById(
+    graphApi: GraphApi,
+    params: { entityId: string },
+  ): Promise<BlockModel> {
+    const entity = await EntityModel.getLatest(graphApi, {
+      // assumption: `accountId` of block is always the workspace account id
+      accountId: workspaceAccountId,
+      entityId: params.entityId,
+    });
+
+    return BlockModel.fromEntityModel(entity);
+  }
+
+  /**
+   * Get a workspace block entity by its component id.
+   *
+   * @param params.componentId - the component id
+   */
+  static async getBlockByComponentId(
```

Comment:
(This has been removed now, comment here is to keep the info)

We can use the query syntax for future entity querires of this shape, but it would require some changes to allow for optional fields
https://github.com/hashintel/hash/commit/b356a08036397c8110b07368346915b104dddd62